homeassistant/components/switch/rainbird.py

```diff
@@ -0,0 +1,224 @@
+        return True
+
+    @property
+    def name(self):
+        """Get the name of the switch."""
+        return self._name
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        self._state = True
+        self._rainbird.start_irrigation(self._zone, self._duration)
+        self._rainbird.setstate(self._zone)
+
+    def turn_off(self, **kwargs):
+        """Turn the switch off."""
+        self._state = False
+        self._rainbird.stop_irrigation()
+        self._rainbird.setstate(0)
+
+    def get_device_status(self):
+        """Get the status of the switch from Rain Bird Controller."""
+        return self._rainbird.state == self._zone
+
+    def update(self):
+        """Update switch status."""
+        self._state = self.get_device_status()
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+
+class RainbirdDevice(Entity):
+    """Rain Bird Device."""
+
+    _state = -1
+
+    def __init__(self, hass, controller):
+        """Initialize the device."""
+        self.hass = hass
+        self.controller = controller
+        self._name = "Rainbird_Controller"
+        self._stations = {}
+
+        # For automation purposes add 2 services
+        def start_irrigation_call(call):
+            """Start irrigation from service call."""
+            station_id = call.data.get("station_id")
+            duration = call.data.get("duration")
+            if station_id and duration:
+                self.start_irrigation(station_id, duration)
+            else:
+                _LOGGER.warning("Error in start_irrigation call. \
+                    station_id and duration need to be set")
+
+        def stop_irrigation_call(call):
+            """Start irrigation from service call."""
+            self.stop_irrigation()
+
+        hass.services.register(DOMAIN, 'start_irrigation',
+                               start_irrigation_call)
+        hass.services.register(DOMAIN, 'stop_irrigation',
+                               stop_irrigation_call)
+
+    def should_poll(self):
+        """Return True if entity has to be polled for state."""
+        return True
+
+    @property
+    def name(self):
+        """Get the name of the device."""
+        return self._name
+
+    def available(self):
+        """Return True if entity is available."""
+        return self._state != -1
+
+    def setstate(self, state):
+        """Force set the current state value."""
+        self._state = state
+
+    @property
+    def state(self):
+        """Return the state of the entity."""
+        return self._state
+
+    def start_irrigation(self, station_id, duration):
+        """
+        Start Irrigation command towards Rain Bird LNK WiFi stick.
+
+        @param call: should be a home assistant call object with data
+        station for Zone to sprinkle and duration for the time
+        """
+        _LOGGER.info("Requesting irrigation for " +
```

Comment:
Please include less logging and limit logging to info to the really necessary. Note that service calls are already logged